From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 21 Mar 2025 12:34:23 -0500
Subject: [PATCH] Debundle glslang

Allow us to link against the system glslang. qtshadertools hasn't caught up to the fact that glslang now has a stable public ABI

Origin: vendor
Forwarded: not-needed
Last-Update: 2025-03-21
---
 CMakeLists.txt                        | 1 +
 src/CMakeLists.txt                    | 2 +-
 src/shadertools/CMakeLists.txt        | 7 +++----
 src/shadertools/qspirvcompiler.cpp    | 4 +---
 src/shadertools/qspirvshaderremap.cpp | 2 +-
 5 files changed, 7 insertions(+), 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e68166a..8b2f43a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,6 +15,7 @@ project(QtShaderTools
 
 find_package(Qt6 ${PROJECT_VERSION} CONFIG REQUIRED COMPONENTS BuildInternals Core)
 find_package(Qt6 ${PROJECT_VERSION} QUIET CONFIG OPTIONAL_COMPONENTS Gui)
+find_package(glslang CONFIG REQUIRED)
 qt_internal_project_setup()
 
 if(INTEGRITY)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index cb0e25d..6a22710 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -3,6 +3,6 @@
 
 # Generated from src.pro.
 
-add_subdirectory(glslang)
+# add_subdirectory(glslang)
 add_subdirectory(SPIRV-Cross)
 add_subdirectory(shadertools)
diff --git a/src/shadertools/CMakeLists.txt b/src/shadertools/CMakeLists.txt
index 04371a5..4e78f33 100644
--- a/src/shadertools/CMakeLists.txt
+++ b/src/shadertools/CMakeLists.txt
@@ -21,15 +21,14 @@ qt_internal_add_module(ShaderTools
         QT_BUILD_SHADERTOOLS_LIB
     INCLUDE_DIRECTORIES
         ../3rdparty/SPIRV-Cross
-        ../3rdparty/glslang
     LIBRARIES
         Qt::GuiPrivate
 # special case begin
         Qt::BundledSpirv_Cross
-        Qt::BundledGlslang_Glslang
-        Qt::BundledGlslang_Spirv
-        Qt::BundledGlslang_Osdependent
 # special case end
+        glslang::glslang
+        glslang::glslang-default-resource-limits
+        glslang::SPVRemapper
     PUBLIC_LIBRARIES
         Qt::Core
         Qt::Gui
diff --git a/src/shadertools/qspirvcompiler.cpp b/src/shadertools/qspirvcompiler.cpp
index 4e97d48..b6fefb8 100644
--- a/src/shadertools/qspirvcompiler.cpp
+++ b/src/shadertools/qspirvcompiler.cpp
@@ -10,7 +10,7 @@ QT_WARNING_PUSH
 QT_WARNING_DISABLE_GCC("-Wsuggest-override")
 #include <glslang/Public/ShaderLang.h>
 #include <glslang/Public/ResourceLimits.h>
-#include <SPIRV/GlslangToSpv.h>
+#include <glslang/SPIRV/GlslangToSpv.h>
 QT_WARNING_POP
 
 //#define TOKENIZER_DEBUG
@@ -47,8 +47,6 @@ bool QSpirvCompilerPrivate::readFile(const QString &fn)
     return true;
 }
 
-using namespace QtShaderTools;
-
 class Includer : public glslang::TShader::Includer
 {
 public:
diff --git a/src/shadertools/qspirvshaderremap.cpp b/src/shadertools/qspirvshaderremap.cpp
index e45161d..d8d4bfe 100644
--- a/src/shadertools/qspirvshaderremap.cpp
+++ b/src/shadertools/qspirvshaderremap.cpp
@@ -3,7 +3,7 @@
 
 #include "qspirvshaderremap_p.h"
 
-#include <SPIRV/SPVRemapper.h>
+#include <glslang/SPIRV/SPVRemapper.h>
 
 QT_BEGIN_NAMESPACE
 
